nca/NetworkConfig/NetworkConfigQuery.py

```diff
@@ -61,6 +64,49 @@ def policy_title(policy):
         """
         return f'{policy.policy_type_str()} {policy.full_name()}'
 
+    # following methods for writing query output in the required format
+    def write_query_output(self, query_answer, explanation_header='', yaml_explanation_descriptions=None, query_result=None):
+        """
+        calls the relevant method to write the query output in the required format
+        :param QueryAnswer query_answer: the query answer
+        :param str explanation_header: the txt message that describes the query's output explanation
+        :param list yaml_explanation_descriptions: [optional] the txt messages of explanations for yaml format,
+        relevant for TwoNetworkConfigsQuery classes
+        :param int query_result: [optional] the numerical result of the query, relevant for TwoNetworkConfigsQuery classes
+        """
+        # compatibility for the called methods
+        yaml_description = yaml_explanation_descriptions or explanation_header
+        numerical_result = query_result if query_result is not None else int(query_answer.numerical_result)
```

Comment:
Why not having the correct/final `numerical_result `within `query_answer`?
One option could be to re-assign `query_answer.numerical_result` with `query_result` if exists.
A possibly better option would be that all queries return `query_answer `that already contains the correct `numerical_result`.
In any case , `write_yaml_output `does not need this extra argument if it can be used from  `query_answer`.

Reply:
agree, this actually  better . I didn't do that in order not to change the old `exec`. 
but it seems that changes on "building" the `QueryAnswer ` fields in `exec()` methods may help a lot to get a cleaner code